File: examples/ruby/add-data-output/DataOutputTest.runar.rb
require 'runar'

# DataOutputTest -- Exercises add_data_output alongside state continuation.

class DataOutputTest < Runar::StatefulSmartContract
  prop :count, Bigint

  def initialize(count)
    super(count)
    @count = count
  end

  # Increment the counter and attach an arbitrary data output whose bytes
  # are committed to by the state continuation hash.
  runar_public payload: ByteString
  def publish(payload)
    @count = @count + 1
    add_data_output(0, payload)
  end
end

File: examples/rust/add-data-output/DataOutputTest.runar.rs
use runar::prelude::*;

/// DataOutputTest -- Exercises add_data_output alongside state continuation.
#[runar::contract]
pub struct DataOutputTest {
    pub count: Bigint,
}

impl DataOutputTest {
    /// Increment the counter and attach an arbitrary data output whose bytes
    /// are committed to by the state continuation hash.
    pub fn publish(&mut self, payload: ByteString) {
        self.count = self.count + 1;
        self.add_data_output(0, payload);
    }
}
